modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1581,7 +1581,7 @@ private boolean tryScrollOneCell(int targetIndex, boolean downOrRight) {
     public void scrollToTop(int index) {
         boolean posSet = false;
 
-        if (index >= getCellCount() - 1) {
+        if (index > getCellCount() - 1) {
```

Comment:
I agree that the previous code violates the contract that the flow should show the specified cell aligned to the top.

I wonder though if this test is needed at all. What is the goal of providing an index that is larger than getCellCount() -1?

Reply:
This part was originally copied from the 'VirtualContainerBase'. There was this comment: _special-case the 'greater than row count' condition to have it be perfectly at position 1_.

Maybe to avoid throwing the IndexOutOfBoundsException or simply to scroll to the end.

Reply:
I am ok with this behavior, but then it should be specified in the javadoc for `@parameter index` that the value can be greater than row count, and that in that case the view will be positioned at the very end.
@kevinrushforth do you have an opinion on this?
This is unrelated to the current fix, so that should not stop this PR.

Reply:
This seems reasonable to me, too. And I agree that it should be specified via a follow-up issue, which will need a CSR.